cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -251,11 +247,25 @@ private static Network getNetworkInfo(String project, String network, Compute co
     return networkObj;
   }
 
+  /**
+   * Extracts and returns the zone name from the given full zone URI.
+   */
+  private static String getZone(String zoneUri) {
+    int idx = zoneUri.lastIndexOf("/");
+    if (idx <= 0) {
+      throw new IllegalArgumentException("Invalid zone uri " + zoneUri);
+    }
+    return zoneUri.substring(idx + 1);
+  }
+
   private static ClusterControllerClient getClusterControllerClient(DataprocConf conf) throws IOException {
     CredentialsProvider credentialsProvider = FixedCredentialsProvider.create(conf.getDataprocCredentials());
 
+    String regionalEndpoint = conf.getRegion() + "-dataproc.googleapis.com:443";
```

Comment:
does the global endpoint also fit this pattern?

Reply:
No. The global endpoint is intentionally removed from the list.